scripts/ddd/ddd

```diff
@@ -95,7 +95,11 @@ restore_mapping() {
 
 	# If we have stored USB_MAP, reload it
 	if [ -f ${DDD_MAP} ]; then
-		source ${DDD_MAP}
+        source ${DDD_MAP}
```

Comment:
I think when the file is sourced inside the function though, it declares a new USB_MAP variable local to the restore_mapping function, inaccessible outside the restore function. When I tried to read any values from it that should have been stored, I could read them in the function but not outside.
Also, thanks so much for writing all this! I was very impressed by the quality and thought I haven't seen often in Bash scripts :D